osu.Game/Overlays/BeatmapSet/HeaderButton.cs

```diff
@@ -2,44 +2,31 @@
 // Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
 
 using osu.Framework.Graphics;
-using osu.Framework.Graphics.Containers;
-using osu.Framework.Graphics.Shapes;
 using osu.Game.Graphics;
-using osu.Game.Graphics.Backgrounds;
-using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.UserInterface;
+using osu.Framework.Allocation;
 
 namespace osu.Game.Overlays.BeatmapSet
 {
-    public class HeaderButton : OsuClickableContainer
+    public class HeaderButton : TriangleButton
     {
-        private readonly Container content;
-
-        protected override Container<Drawable> Content => content;
 
         public HeaderButton()
         {
-            CornerRadius = 3;
-            Masking = true;
+            Height = 0;
+            RelativeSizeAxes = Axes.Y;
+        }
 
-            InternalChildren = new Drawable[]
-            {
-                new Box
-                {
-                    RelativeSizeAxes = Axes.Both,
-                    Colour = OsuColour.FromHex(@"094c5f"),
-                },
-                new Triangles
-                {
-                    RelativeSizeAxes = Axes.Both,
-                    ColourLight = OsuColour.FromHex(@"0f7c9b"),
-                    ColourDark = OsuColour.FromHex(@"094c5f"),
-                    TriangleScale = 1.5f,
-                },
-                content = new Container
-                {
-                    RelativeSizeAxes = Axes.Both,
-                },
-            };
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            Masking = true;

```

Comment:
is this required?

Reply:
Can't change the Triangles, BGColor, etc. in the constructor, since they are initialized in TriangleButtons load method.

Reply:
the masking link

Reply:
Although the Masking and the CornerRadius do not seem to do anything... Should I remove them?

Reply:
yes please